cfgmgr/intfmgr.cpp

```diff
@@ -71,51 +98,139 @@ bool IntfMgr::isIntfStateOk(const string &alias)
 
     return false;
 }
-void IntfMgr::doTask(Consumer &consumer)
+
+bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
+        const vector<FieldValueTuple>& data,
+        const string& op)
 {
     SWSS_LOG_ENTER();
 
-    auto it = consumer.m_toSync.begin();
-    while (it != consumer.m_toSync.end())
+    string alias(keys[0]);
+    string vrf_name = "";
+
+    for (auto idx : data)
     {
-        KeyOpFieldsValuesTuple t = it->second;
+        const auto &field = fvField(idx);
+        const auto &value = fvValue(idx);
+        if (field == "vnet_name" || field == "vrf_name")
+        {
+            vrf_name = value;
+        }
+    }
 
-        vector<string> keys = tokenize(kfvKey(t), config_db_key_delimiter);
+    if (op == SET_COMMAND)
+    {
+        /*
+         * Don't proceed if port/LAG/VLAN is not ready yet.
+         * The pending task will be checked periodically and retried.
+         * TODO: Subscribe to stateDB for port/lag/VLAN state and retry
+         * pending tasks immediately upon state change.
+         */
+        if (!isIntfStateOk(alias))
+        {
+            SWSS_LOG_DEBUG("Interface is not ready, skipping %s", alias.c_str());
+            return false;
+        }
 
-        if (keys.size() != 2)
+        if (!isIntfStateOk(vrf_name))
```

Comment:
vrf_name is optional and can be empty here. You would want to check if it is empty. User may just want to configure "mtu" for the interface.

Reply:
Done